### PR TITLE
add notification channel for home assistant

### PR DESCRIPTION
**Pull request by OptimusGREEN**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]:
- [x] I have read and understand the pull request rules.

# Description

This adds a notification channel for the home assistant notification method so android users can set uptime kuma specific notification sounds etc...

[Notification Channels](https://companion.home-assistant.io/docs/notifications/notifications-basic/#notification-channels)
## Type of change

Please delete any options that are not relevant.

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [ ] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [x] My changes generates no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

Please do not use any external image service. Instead, just paste in or drag and drop the image here, and it will be uploaded automatically.

### CODE REVIEW

**Comment by OptimusGREEN** [comment]
I still recieve the notification on my mac, the notification channel is just ignored.

**Review by rmtsrc — APPROVED** [review]
Looks like it works for me and I'm able to customize how Uptime Kuma notifications appear (vibration, sounds etc.) separately from Home Assistant notifications 👍







```log
home-assistant  | 2024-03-01 21:43:37.102 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: domain=notify, service=my-mobile, service_data=title=Uptime Kuma, message=My Home Assistant Alert Testing, data=name=undefined, status=undefined, channel=uptime-kuma>
home-assistant  | 2024-03-01 21:43:37.102 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: domain=notify, service=mobile_android_app, service_data=message=My Home Assistant Alert Testing, title=Uptime Kuma, data=name=undefined, status=undefined, channel=uptime-kuma>
home-assistant  | 2024-03-01 21:43:37.102 DEBUG (Recorder) [homeassistant.components.recorder.core] Processing task: <Event call_service[L]: domain=notify, service=my-mobile, service_data=title=Uptime Kuma, message=My Home Assistant Alert Testing, data=name=undefined, status=undefined, channel=uptime-kuma>
home-assistant  | 2024-03-01 21:43:37.103 DEBUG (Recorder) [homeassistant.components.recorder.core] Processing task: <Event call_service[L]: domain=notify, service=mobile_android_app, service_data=message=My Home Assistant Alert Testing, title=Uptime Kuma, data=name=undefined, status=undefined, channel=uptime-kuma>
```

Looking at the docs for the Android Home Assistant app we could even customize the notification icon to be the Uptime Kuma logo (rather than Gone Assistant) which would make it a little easier to distinguish in the notification list

**Comment by CommanderStorm** [comment]
@rmtsrc could you link the relevant part of the docs? What would need to be changed?

File: server/notification-providers/home-assistant.js [hunk 1]
```diff
@@ -21,6 +21,7 @@ class HomeAssistant extends NotificationProvider {
                     ...(notificationService !== "persistent_notification" && { data: {
                         name: monitor?.name,
                         status: heartbeat?.status,
+                        channel: "uptime-kuma",
```
Comment (CommanderStorm) [inline server/notification-providers/home-assistant.js:24]:
```suggestion
                        channel: "uptime-kuma",
                        icon_url: "https://github.com/louislam/uptime-kuma/blob/master/public/icon-192x192.png?raw=true"
```

**Comment by CommanderStorm** [comment]
@OptimusGREEN is there a reason that this PR is a draft?

**Comment by OptimusGREEN** [comment]
Hi, yes I'm pretty sure the contributing guide said to make PR's drafts?

**Comment by CommanderStorm** [comment]
> I'm pretty sure the contributing guide said to make PR's drafts?

Yea, we changed said wording to make it more clear what the intent of that was ^^
The `create an empty draft pull request or open an issue, so we can have a discussion first` is only required for features to prevent wasted work on both sides (contributor/maintainer)

I am reading your comment as "This would be ready to go" and have changed this to be `Ready for Review`.

**Comment by OptimusGREEN** [comment]
Ahhh, makes sense. Yup I'm confident it's ready to go thanks.

**Comment by OptimusGREEN** [comment]
> @rmtsrc Could you test if this works as intended? (I would feel way more confident in this not breaking peoples setup if you present a screenshot ^^) You can test-drive PRs via this mechanism: https://github.com/louislam/uptime-kuma/wiki/Test-Pull-Requests

Just tested this and it adds an icon but doesn't replace the home assistant icon.

Screenshot_20240311_210605_Nova7

**Comment by CommanderStorm** [comment]
Unsure if we should keep this…
What the other people in this thread think?

**Comment by OptimusGREEN** [comment]
It looks a bit strange. But I'm easy either way.

File: server/notification-providers/home-assistant.js [hunk 2]
```diff
@@ -21,6 +21,8 @@ class HomeAssistant extends NotificationProvider {
...
+                        icon_url: "https://github.com/louislam/uptime-kuma/blob/master/public/icon-192x192.png?raw=true"
```
[the remark this answers is not in the document]
Comment (OptimusGREEN) [reply server/notification-providers/home-assistant.js:25]: Yeah, looks much better

**Comment by CommanderStorm** [comment]
I am taking the sentiment as "keep the `icon_url`".
Is this correct?

**Comment by OptimusGREEN** [comment]
> I am taking the sentiment as "add the `icon_url`".
> Is this correct?

Yeah 👍🏻

**Comment by CommanderStorm** [comment]
Thanks for the enhancement!
All changes in this PR are small and uncontroversial
⇒ merging with junior maintainer approval

**Comment by Saibamen** [comment]
Should be backported to `1.23.X` branch

**Comment by CommanderStorm** [comment]
> Should be backported to `1.23.X` branch

I consider this more to be a feature than a bugfix
=> leaving this in v2.0 would be fine by me

**Pull request merged**